pkg/agent/sysadvisor/plugin/inference/modelresultfetcher/borwein/borwein.go

```diff
@@ -193,21 +192,32 @@ func (bmrf *BorweinModelResultFetcher) parseInferenceRespForPods(requestContaine
 			return nil, fmt.Errorf("invalid containerEntries for pod: %s", podUID)
 		}
 
-		results[podUID] = make(map[string]*borweininfsvc.InferenceResult)
+		results[podUID] = make(map[string][]*borweininfsvc.InferenceResult)
 
-		for containerName, result := range containerEntries.ContainerInferenceResults {
-			if result == nil {
+		for containerName, cResults := range containerEntries.ContainerInferenceResults {
+			if cResults == nil {
 				return nil, fmt.Errorf("invalid result for pod: %s, container: %s", podUID, containerName)
 			}
 
-			results[podUID][containerName] = proto.Clone(result).(*borweininfsvc.InferenceResult)
+			results[podUID][containerName] = make([]*borweininfsvc.InferenceResult, len(cResults.InferenceResults))
+
+			for idx, result := range cResults.InferenceResults {
+				if result == nil {
+					continue
+				}
+
+				results[podUID][containerName][idx] = proto.Clone(result).(*borweininfsvc.InferenceResult)
+			}
+			// TODO? should check inference results here?
 		}
 
 		respContainersCnt += len(results[podUID])
 	}
 
+	// This may happen when some containers lack essential features.
+	// So just log warning here.
 	if respContainersCnt != len(requestContainers) {
-		return nil, fmt.Errorf("count of resp containers: %d and request containers: %d are not same",
+		klog.Warningf("count of resp containers: %d and request containers: %d are not same",
```

Comment:
If respContainersCnt != len(requestContainers),  the possibility of abnormal state may increase since denominator may decrease; so we still return error here.

And if some containers lack essential features, we also inference it forcibly.